synapse/storage/databases/main/lock.py

```diff
@@ -112,6 +120,7 @@ def _try_acquire_lock_txn(txn: LoggingTransaction) -> bool:
                             last_renewed_ts = EXCLUDED.last_renewed_ts
                         WHERE
                             worker_locks.last_renewed_ts < ?
+                            OR worker_locks.instance_name = EXCLUDED.instance_name
```

Comment:
Does the `DO UPDATE` above only apply to the row(s?) that's caused a conflict? If not I'm not sure I understand what's happening here.

Reply:
Yup, so it's basically equivalent to doing the `INSERT` if there is no conflict, or applying the `UPDATE SET ... WHERE ...` if there is a conflict (with the special `EXCLUDED.*` table matching the `VALUES` in the `INSERT`).  

I.e. in this specific case we're either inserting a new row, or updating the existing row so long as either a) the lock has timed out OR b) the instance name matches.